### PR TITLE
Add sheep shearing

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

```SUMMARY: Features "Add sheep shearing"```

#### Purpose of change
Adds Sheep Shearing

#### Describe the solution
Remembered that @RosaEscura was working on this a while back, looked at their branch, and it was unfinished and no work had happened on it for a good while, so ive mostly just adapted the existing json entries there, added recipes and the C++ code to the right place, and converted it to a player activity.

#### Describe alternatives you've considered
Didnt make the wool type definable in JSOn for modders to add weird and wonderful creatures, or other such variability, yet, anyway.

Just sheep = wool result so far.
Added a static amount of wool staples  based on an average of the amount a wool-producing sheep might yield ( around 10 lbs apparently )

in the future this could be variable based on species, with a defined itype in the JSOn like milk is for cows etc.

#### Testing
Tried shearing without shears - couldnt.
Tried shearing in winter - couldnt
Sheared with electric shears - activity started, took less amount of time than manual shears, and used charges, activity finished fine.
Sheared with normal shears - activity started and finished fine.
Time to regrow wool is set to season length, debug passing time dosnt actually progress effect timers, so I tested this by setting the effect time to 2 hours, then waiting and could then shear again.

The same logic should apply to an effect timer of 12 weeks.
#### Additional context
N/A

### CODE REVIEW

File: data/json/recipes/other/tool.json [hunk 1]
```diff
@@ -55,6 +55,36 @@
...
+  {
+    "result": "shears",
+    "type": "recipe",
+    "category": "CC_OTHER",
+    "subcategory": "CSC_OTHER_TOOLS",
+    "skill_used": "fabrication",
+    "difficulty": 3,
+    "time": "3 h",
+    "autolearn": true,
+    "reversible": true,
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: Wouldn't make sense for this recipe to be reversible, I think. Breaking down shears back into metal would need at least a hacksaw and at most a forge; same for recipe below.

File: src/monexamine.cpp [hunk 2]
```diff
@@ -117,6 +121,23 @@ bool monexamine::pet_menu( monster &z )
     if( z.has_flag( MF_MILKABLE ) ) {
         amenu.addentry( milk, true, 'm', _( "Milk %s" ), pet_name );
     }
+    if( z.has_flag( MF_SHEARABLE ) ) {
+        bool available = true;
+        if( season_of_year( calendar::turn ) == WINTER ) {
+            amenu.addentry( shear, false, 'S', _( "This animal would freeze if you sheared it now" ) );
```
Comment (Rail-Runner) [inline src/monexamine.cpp]:
```suggestion
            amenu.addentry( shear, false, 'S', _( "This animal would freeze if you sheared it now." ) );
```

File: src/monexamine.cpp [hunk 3]
```diff
@@ -117,6 +121,23 @@ bool monexamine::pet_menu( monster &z )
...
+            available = false;
+        } else if( z.has_effect( effect_sheared ) ) {
+            amenu.addentry( shear, false, 'S', _( "This animal is not ready to shear again yet" ) );
```
Comment (Rail-Runner) [inline src/monexamine.cpp]:
```suggestion
            amenu.addentry( shear, false, 'S', _( "This animal is not ready to shear again yet." ) );
```

File: data/json/recipes/other/tool.json [hunk 4]
```diff
@@ -55,6 +55,36 @@
...
+    "using": [ [ "forging_standard", 4 ] ],
+    "qualities": [ { "id": "ANVIL", "level": 3 }, { "id": "HAMMER", "level": 3 }, { "id": "CHISEL", "level": 3 } ],
+    "tools": [ [ [ "tongs", -1 ] ], [ [ "swage", -1 ] ], [ [ "mold_plastic", -1 ] ] ],
+    "components": [ [ [ "plastic_chunk", 1 ] ], [ [ "scrap", 3 ] ] ]
```
Comment (Rail-Runner) [inline data/json/recipes/other/tool.json]: One lump of steel (which is 1 kg) would better represent the item's current weight than three pieces of scrap metal (150 g). Same for recipe below.

File: data/json/items/tool/pets.json [hunk 5]
```diff
@@ -29,6 +29,42 @@
...
+    "material": [ "steel", "plastic" ],
+    "symbol": ";",
+    "color": "light_gray",
+    "qualities": [ [ "CUT", 1 ], [ "SHEAR", 1 ] ],
+    "flags": [ "STAB" ]
+  },
+  {
+    "id": "elec_shears",
+    "type": "TOOL",
+    "name": "electric shears",
```
Comment (Rail-Runner) [inline data/json/items/tool/pets.json]: Need to specify valid plural name here or it'll be "electric shearss".

File: src/activity_handlers.cpp [hunk 6]
```diff
@@ -1271,6 +1274,39 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
...
+        debugmsg( "shearing item location lost" );
+        return;
+    }
+    const tripoint source_pos = g->m.getlocal( act->coords.at( 0 ) );
+    monster *source_mon = g->critter_at<monster>( source_pos );
+    if( source_mon == nullptr ) {
+        debugmsg( "could not find source creature for shearing" );
+        return;
+    }
+    // 22 wool stables corresponds to an average wool-producing sheep yield of 10 lbs or so
```
Comment (Night-Pryanik) [inline src/activity_handlers.cpp]:
```suggestion
    // 22 wool staples corresponds to an average wool-producing sheep yield of 10 lbs or so
```

File: src/monexamine.cpp (hunk 2 above)
Comment (Night-Pryanik) [inline src/monexamine.cpp]:
```suggestion
            amenu.addentry( shear, false, 'S', _( "This animal would freeze if you shear it now." ) );
```

File: src/monexamine.cpp (hunk 3 above)
Comment (Night-Pryanik) [inline src/monexamine.cpp]:
```suggestion
            amenu.addentry( shear, false, 'S', _( "This animal is not ready to be sheared again yet." ) );
```

Comment (davidpwbrown) [reply src/monexamine.cpp]: addentry strings dont require fullstops.

Comment (davidpwbrown) [reply src/monexamine.cpp]: Or at least, no other addentry strings elsewhere in the code that ive seen, use them.

Comment (davidpwbrown) [reply src/monexamine.cpp]: Though that could be due to conversations like this in the past. no harm in adding them.

File: src/monexamine.cpp [hunk 7]
```diff
@@ -124,6 +128,24 @@ bool monexamine::pet_menu( monster &z )
     if( z.has_flag( MF_MILKABLE ) ) {
         amenu.addentry( milk, true, 'm', _( "Milk %s" ), pet_name );
     }
+    if( z.has_flag( MF_SHEARABLE ) ) {
+        bool available = true;
+        if( season_of_year( calendar::turn ) == WINTER ) {
+            amenu.addentry( shear, false, 'S',
+                            _( "This animal would freeze if you sheared it during Winter." ) );
```
Comment (matskuman5) [inline src/monexamine.cpp]:
```suggestion
                            _( "This animal would freeze if you sheared it during winter." ) );
```

File: src/monexamine.cpp [hunk 8]
```diff
@@ -124,6 +128,24 @@ bool monexamine::pet_menu( monster &z )
     if( z.has_flag( MF_MILKABLE ) ) {
         amenu.addentry( milk, true, 'm', _( "Milk %s" ), pet_name );
     }
+    if( z.has_flag( MF_SHEARABLE ) ) {
+        bool available = true;
+        if( season_of_year( calendar::turn ) == WINTER ) {
+            amenu.addentry( shear, false, 'S',
+                            _( "This animal would freeze if you sheared it during winter." ) );
```
Comment (Night-Pryanik) [inline src/monexamine.cpp]:
```suggestion
                            _( "This animal would freeze if you shear it during winter." ) );
```

**Comment by RosaEscura** [comment]
<3

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/electric-shears-inconsistent/23351/2

**Pull request merged**